spanner/api/Spanner.Samples/TransactionTagAsync.cs

```diff
@@ -0,0 +1,86 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_set_transaction_tag]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class TransactionTagAsyncSample
+{
+    public class Venue
+    {
+        public long VenueId { get; set; }
+        public string VenueName { get; set; }
+        public long Capacity { get; set; }
+    }
+    
+    public async Task<List<Venue>> TransactionTagAsync(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+        using var connection = new SpannerConnection(connectionString);
+        await connection.OpenAsync();
+
+        return await connection.RunWithRetriableTransactionAsync(async transaction =>
+        {
+            var venues = new List<Venue>();
+            // Sets the transaction tag to "app=concert,env=dev".
+            // This transaction tag will be applied to all the individual operations inside
+            // the transaction.
+            transaction.Tag = "app=concert,env=dev";
+            var selectCommand = connection.CreateSelectCommand(
+                @"SELECT VenueId, VenueName, Capacity
+                  FROM Venues
+                  WHERE OutdoorVenue = @outdoorVenue",
+                new SpannerParameterCollection
+                {
+                    new SpannerParameter("outdoorVenue", SpannerDbType.Bool, false)
+                });
+            selectCommand.Transaction = transaction;
+            // Sets the request tag to "app=concert,env=dev,action=select".
+            // This request tag will only be set on this request.
+            selectCommand.Tag = "app=concert,env=dev,action=select";
+            using var reader = await selectCommand.ExecuteReaderAsync();
+            while (await reader.ReadAsync())
+            {
+                var venue = new Venue
+                {
+                    VenueId = reader.GetFieldValue<long>("VenueId"),
+                    VenueName = reader.GetFieldValue<string>("VenueName"),
+                    Capacity = reader.GetFieldValue<long>("Capacity")
+                };
+                venues.Add(venue);
+                // Update the capacity of the venue.
+                venue.Capacity /= 4;
+                var updateCommand = connection.CreateDmlCommand(
+                    "UPDATE Venues SET Capacity = @capacity WHERE VenueId = @venueId",
+                    new SpannerParameterCollection
+                    {
+                        {"capacity", SpannerDbType.Int64, venue.Capacity},
+                        {"venueId", SpannerDbType.Int64, venue.VenueId}
+                    });
```

Comment:
It feels wrong to go through all this trouble when we can just do `UPDATE Venues SET Capacity = Capacity/4`. This is the kind of sample that promotes bad practices. I know we had a few of these from before but we have fixed them. If we want to show two different command tags being set, can we do an update and an insert instead of the unnecesary select?

Reply:
Good point. Once again this is something that should (probably) be kept equal across all programming languages, so it's not really my call. @skuruppu Any thoughts on this?

Reply:
Hi all, I wrote the samples for this. I was avoiding inserts as I was worried they may affect other samples which isn't, admittedly, isn't a great reason :sweat_smile:  I am happy to update the samples to avoid promoting bad practices :+1: 

Are there practices I should avoid when rewriting the samples to include an `UPDATE` and an `INSERT` statement ?

Reply:
Thanks for the quick response.

Regarding good practices for `UPDATE` and `INSERT` statements: I would say 'prefer parameters above literals for constant values' in the statements. So `INSERT INTO Foo (Id, Value) VALUES (@id, @value)` above `INSERT INTO Foo (Id, Value) VALUES (1, 'test')`.

Reply:
Thanks for the quick response.

Regarding good practices for `UPDATE` and `INSERT` statements: I would say 'prefer parameters above literals for constant values' in the statements. So `INSERT INTO Foo (Id, Value) VALUES (@id, @value)` above `INSERT INTO Foo (Id, Value) VALUES (1, 'test')`.

Reply:
+1 to what @olavloite said. And in general, don't do in a sample anything you wouldn't do in your own application (as the unnecesary select here). I do understand the worry about the inserts interfering, we have sometimes struggle with that as well, but it's better to show as real life code as possible :)! Thanks again.

I'll wait till the new sample is out then?

Reply:
The specs and the sample have been updated. PTAL.

@larkee I've had to make two small changes to the code that deviate from the exact sample code:
- Use `DIV(Capacity, 4)` instead of `Capacity / 4` as Capacity is INT64.
- Add a value for `LastUpdatedTime` in the insert statement, as it is a required column.

(The same has also been applied to the [NodeJS sample](https://github.com/googleapis/nodejs-spanner/pull/1450/files))